### PR TITLE
API changes

### PR DESCRIPTION
**Pull request by cboesch-gpsw**
Address 2 points from https://github.com/gopro/gopro-lib-node.gl/issues/50

### CODE REVIEW

**Comment by mbouron-gpsw** [comment]
Can you state in the commit description of "api: remove va-arg constructor mechanism", that the API temporarily unsafe to use as there is no more checks for x,y,z in Compute, no check for geometry in Render, ... ?

File: libnodegl/node_camera.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ struct camera_priv {
 
 #define OFFSET(x) offsetof(struct camera_priv, x)
 static const struct node_param camera_params[] = {
-    {"child", PARAM_TYPE_NODE, OFFSET(child), .flags=PARAM_FLAG_CONSTRUCTOR,
+    {"child", PARAM_TYPE_NODE, OFFSET(child), .flags=PARAM_FLAG_NOT_NULL,
```
Comment (mbouron-gpsw) [inline libnodegl/node_camera.c:64]: PARAM_FLAG_NON_NULL instead of PARAM_FLAG_NOT_NULL ?

File: libnodegl/node_triangle.c [hunk 2]
```diff
@@ -32,11 +32,14 @@
...
+    {"edge0", PARAM_TYPE_VEC3, OFFSET(triangle_edges[0]),
+              {.vec={1.0, 0.0, 0.0}},
               .desc=NGLI_DOCSTRING("first edge coordinate of the triangle")},
-    {"edge1", PARAM_TYPE_VEC3, OFFSET(triangle_edges[3]), .flags=PARAM_FLAG_CONSTRUCTOR,
+    {"edge1", PARAM_TYPE_VEC3, OFFSET(triangle_edges[3]),
+              {.vec={0.0, 1.0, 0.0}},
               .desc=NGLI_DOCSTRING("second edge coordinate of the triangle")},
-    {"edge2", PARAM_TYPE_VEC3, OFFSET(triangle_edges[6]), .flags=PARAM_FLAG_CONSTRUCTOR,
+    {"edge2", PARAM_TYPE_VEC3, OFFSET(triangle_edges[6]),
+              {.vec={0.0, 0.0, 0.0}},
```
Comment (mbouron-gpsw) [inline libnodegl/node_triangle.c:42]: Maybe center it around 0,0 ((1,-1), (1, 0), (-1,1)) or a scaled down version (still centered around 0) ?

**Comment by cboesch-gpsw** [comment]
> Can you state in the commit description of "api: remove va-arg constructor mechanism", that the API temporarily unsafe to use as there is no more checks for x,y,z in Compute, no check for geometry in Render, ... ?

Sure, done.

File: libnodegl/node_camera.c (hunk 1 above)
Comment (cboesch-gpsw) [reply libnodegl/node_camera.c]: changed.

File: libnodegl/node_triangle.c (hunk 2 above)
Comment (cboesch-gpsw) [reply libnodegl/node_triangle.c]: ok

**Pull request merged**